packages/hoppscotch-ui/src/components/smart/Table.vue

```diff
@@ -0,0 +1,129 @@
+          :class="yBorder ? 'divide-x divide-divider' : ''"
+        >
+          <td
+            v-for="(data, colIndex) in item"
+            :key="colIndex"
+            @click="emit('goToDetails', item)"
+            class="max-w-40"
+            :class="cellStyles"
+          >
+            <div class="flex items-center">
+              <!-- Column with Badge -->
+              <div
+                v-if="
+                  colIndex.toString() === badgeColName &&
+                  badgeRowIndices?.includes(rowIndex)
+                "
+              >
+                <div class="flex flex-col truncate">
+                  <span v-if="data" class="mt-1 truncate whitespace-normal">
+                    {{ data }}
+                  </span>
+
+                  <span
+                    class="text-xs font-medium px-3 py-1 my-1 rounded-full bg-green-900 text-green-300 w-min"
+                  >
+                    {{ badgeName }}
+                  </span>
+                </div>
+              </div>
+
+              <!-- Column with Subtitles -->
+              <div
+                v-else-if="subtitleColumns.includes(colIndex.toString())"
+                class="flex flex-col truncate"
+              >
+                <span v-if="data" class="truncate">
+                  {{ data }}
+                </span>
+                <span v-else> - </span>
+
+                <div v-for="item in subtitles">
+                  <div
+                    v-if="item.colName === colIndex.toString()"
+                    class="text-gray-400 text-tiny"
+                  >
+                    <span>
+                      <span>
+                        {{ itemSubtitle(item.subtitle, rowIndex) }}
+                      </span>
+                    </span>
+                  </div>
+                </div>
+              </div>
+
+              <!-- Column with no subtitle or badge -->
+              <div v-else class="flex flex-col truncate">
+                <span v-if="data" class="truncate">
+                  {{ data }}
+                </span>
+                <span v-else> - </span>
+              </div>
+            </div>
+          </td>
+
+          <slot name="action" :item="item"></slot>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+
+<script lang="ts" setup generic="Item extends Record<string, unknown>">
+import { computed } from "vue"
+
+const props = defineProps<{
+  /** Whether to show the vertical border between columns */
+  yBorder?: boolean
+  /**  The list of items to be displayed in the table */
+  list: Item[]
+  /** The headings of the table */
+  headings: string[]
+  /** The styles to be applied to the table cells */
+  cellStyles?: string
+  /** The name of the badge */
+  badgeName?: string
+  /** The indices of the rows that needs to have a badge */
+  badgeRowIndices?: (number | undefined)[]
+  /** The index of the column that needs to have a badge */
+  badgeColName?: string
```

Comment:
Since it's a genertic table component. badge related props shouldn't be there. All table are not gonna have badge for its specific cell